dask/array/routines.py

```diff
@@ -546,7 +547,7 @@ def roll(array, shift, axis=None):
 
         result = concatenate([result[sl1], result[sl2]], axis=i)
 
-    result = result.reshape(array.shape)
```

Comment:
Note that rechunking can be expensive.  I'm inclined not to rechunk if not necessary.

Reply:
Or maybe I should instead ask "why was rechunking important here?"